### PR TITLE
Shells

### PR DESCRIPTION
**Pull request by AndersonTorres**
MirBSD and Public Domain Korn shells.

### CODE REVIEW

File: pkgs/shells/pdksh/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    url = "http://www.cs.mun.ca/~michael/pdksh/files/${name}.tar.gz";
+    sha256 = "19mzkfmj3ms8sp9m8f6fyba0am6fxnr538xw6b3fv4gja3fvq5db";
+  };
+
+  buildInputs = [ ];
+
+  # Traditionally, PDKSH is used as "pdksh" rather than "ksh"
+  configureFlags = [ "--program-prefix=pd" ];
+
+  patches = [
```
Comment (peti) [inline pkgs/shells/pdksh/default.nix]: Wow, that's a lot of extensive patches. Is there any particular reason why these changes aren't included upstream?

Also, can we maybe download the diff files instead of checking them into the Nixpkgs repository?

Comment (matthiasbeyer) [inline pkgs/shells/pdksh/default.nix]: These patches should really go into upstream, they look like behaviour-changing to me and I think we shouldn't patch software that much in a distribution repository!

Comment (AndersonTorres) [inline pkgs/shells/pdksh/default.nix]: Regularly I watch AUR or regular Archlinux PKGBUILDs in order to be more acquainted with the software packaging. In AUR, there is a big fat patch, and I have split them in my testing. But the official homepage has a patch too:
http://www.cs.mun.ca/~michael/pdksh/files/

pdksh is a bit outdated (2008?), so I think it is more secure and sensible if we maintain our particular set.

Comment (peti) [inline pkgs/shells/pdksh/default.nix]: I'd very much prefer to have those patch files downloaded from the official site (like our `bash` build and many other packages do it). The Nixpkgs repository is intended for collecting build expressions, and we'd rather not include patches unless it's absolutely necessary. This seems to be no such case, because the patches are readily available elsewhere.

Comment (AndersonTorres) [inline pkgs/shells/pdksh/default.nix]: I will update with the upstream patches, then.

Comment (AndersonTorres) [inline pkgs/shells/pdksh/default.nix]: Bad news: these patches are really necessary.
I have tested the upstream patches, and they fail in a subtle file, siglist.sh.
So, we really need to maintain local patches in this case.

Comment (cstrahan) [inline pkgs/shells/pdksh/default.nix]: Or we could just patch the patch files.

I kid :).

**Comment by peti** [comment]
Have you tried contacting Michael Rendell <michael@cs.mun.ca> to get those patches included in the upstream distribution?

**Comment by AndersonTorres** [comment]
Well, I will try it now - but pdksh is a bit old, and the last update is from 2008. I don't think he is interested in maintain pdksh, even because there are so many pdksh forks floating around (the mksh being the best IMHO).

Until I obtain some response, I will just drop pdksh and its funny patches.

**Comment by AndersonTorres** [comment]
1 - Why Travis is failing again?
2 - Can we accept mksh now? :)
